src/privacy_mode/win_input.rs

```diff
@@ -22,14 +23,18 @@ const GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS: u32 = 4;
 const WM_USER_EXIT_HOOK: u32 = WM_USER + 1;
 
 lazy_static::lazy_static! {
-    static ref CUR_HOOK_THREAD_ID: Mutex<DWORD> = Mutex::new(0);
+    /// thread id of the currently attached to thread (non-zero)
+    static ref CUR_HOOK_THREAD_ID: Mutex<Option<NonZero<DWORD>>> = Mutex::new(None);
 }
 
+/// Hook keyboard and mouse to the current thread
 fn do_hook(tx: Sender<String>) -> ResultType<(HHOOK, HHOOK)> {
```

Comment:
We validate the HHOOK pointers to not be null, so we could make this return `NonNull<HHOOK>`. For example we currently recheck that the keyboard hook isn't null for a second time in `fn hook()` even though we have already validated it.